ocaml/xapi/xapi_cluster_helpers.ml

```diff
@@ -57,18 +57,13 @@ let get_operation_error ~__context ~self ~op =
   in
   current_error
 
+let assert_allowed_during_rpu __context = function
+  | `add | `remove | `destroy ->
+    Helpers.assert_rolling_upgrade_not_in_progress ~__context
+  | _ -> ()
+
 let assert_operation_valid ~__context ~self ~op =
-  begin match (Helpers.rolling_upgrade_in_progress ~__context), op with
-    | true, (`add | `remove | `destroy) ->
-      raise Api_errors.(Server_error (not_supported_during_upgrade, []))
-    | _, `remove -> ()
-    | _, _ -> begin
-      match Xapi_clustering.find_cluster_host ~__context ~host:Helpers.(get_localhost ~__context) with
-      | Some self ->
-        if not (Db.Cluster_host.get_joined ~__context ~self)
-        then raise Api_errors.(Server_error (cluster_host_not_joined, [ Ref.string_of self ]))
-      | None -> () end
-  end;
+  assert_allowed_during_rpu __context op;
```

Comment:
Same comment as below, now that you refactored it, its obvious that this is in the wrong place, please move it inside `get_operation_error`.